neuralop/models/codano.py

```diff
@@ -0,0 +1,371 @@
+from ..layers.resample import resample
+from ..layers.embeddings import GridEmbedding2D, GridEmbeddingND
+
+class CODANO(nn.Module):
+    """
+    Parameters
+    ---
+    output_variable_codimension : int
+        number of output channels corresponding to each variable
+    hidden_variable_codimension : int
+        number of hidden channels corresponding to each variable
+    lifting_variable_codimension : int
+        number of hidden channels in lifting/projection module for each variable
+    use_positional_encoding : bool
+        whether to use varibale specific positional encoding
+    positional_encoding_dim : int
+        dimension (number of channels) of positional encoding
+    static_channel_dim : int
+        number of channels in static channel. These channels contains additional static information of the PDE such as boundary conditions.
+    positional_encoding_modes : list
+        number of Fourier modes used in positional encoding
+    n_variables : int
+        number of variables
+    use_horizontal_skip_connection : bool
+        whether to use horizontal skip connections
+    horizontal_skips_map : dict
+                    a map {...., b: a, ....} denoting horizontal skip connection from a-th layer to
+                    b-th layer.
+                    Example: For a 5 layer UNO architecture, the skip connections can be
+                    horizontal_skips_map ={4:0,3:1}
+    n_layers : int
+        number of layers
+    n_modes : list
+        Number of Fourier Modes to use in integral operations in the CoDA-NO block (along each dimension).
+        Example: For a five layer CODANO with 2D input the n_modes can be: [[16,16],[16,16],[16,16],[16,16],[16,16]]
+    scalings : list
+        the output scaling factor for each layer along each dimension. 
+        For example, for a 2D input, the scalings can be [[1,1],[0.5,0.5],[1,1], [2,2], [1,1]]
+    n_heads : list
+        number of attention heads for each layer
+    attention_scalings : list
+        scaling factor in attention mechanism. For example, for a 5 layer CoDA-NO, the attention_scalings can be [0.5, 0.5, 0.5, 0.5, 0.5].
+    conv_module : nn.Module
+        convolution module to use in the CoDA-NO block
+    nonlinear_attention : bool
+        whether to use non-linear attention mechanism
+    non_linearity : nn.Module
+        non-linearity to use in the CoDA-NO block
+    attention_token_dim : int
+        number of channels for each token for attention mechanism
+    per_channel_attention : bool
+        whether to use per channel attention mechanism
+    layer_kwargs : dict
+        additional arguments for the CODABlocks
+    projection : bool
+        whether to use projection module
+    lifting : bool
+        whether to use lifting module
+    domain_padding : float
+        padding factor for domain padding
+    domain_padding_mode : str
+        domain padding mode (one-sided or two-sided)
+    enable_cls_token : bool
```

Comment:
And could you describe what `CLS` is?

Reply:
Clarified.

Reply:
I don't know if this is clear enough. What exactly is a learnable class token?